core/src/main/java/com/sequenceiq/cloudbreak/service/credential/CredentialService.java

```diff
@@ -72,135 +76,111 @@
-        notification.setCloud(credential.cloudPlatform());
-        notificationSender.send(new Notification<>(notification));
-    }
-
-    public Credential getPublicCredential(String name, IdentityUser user) {
-        return Optional.ofNullable(credentialRepository.findOneByName(name, user.getAccount()))
-                .orElseThrow(accessDenied(String.format("Access is denied: Credential not found by name '%s'", name)));
+    public Credential create(Credential credential) {
+        credential.setOrganization(getDefaultOrg());
+        return create(credential, getDefaultOrg().getId());
     }
 
-    public Credential getPrivateCredential(String name, IdentityUser user) {
-        return Optional.ofNullable(credentialRepository.findByNameInUser(name, user.getUserId()))
-                .orElseThrow(accessDenied(String.format("Access is denied: Credential not found by name '%s'.", name)));
+    @Override
+    public Credential create(Credential credential, Long orgId) {
+        LOGGER.debug("Creating credential for organization: {}", getOrganizationService().get(orgId).getName());
+        checkCredentialCloudPlatform(credential.cloudPlatform());
+        Credential created = super.create(credentialAdapter.init(credential), orgId);
+        sendCredentialNotification(credential, ResourceEvent.CREDENTIAL_CREATED);
+        return created;
     }
 
-    public void delete(Long id, IdentityUser user) {
-        Credential credential = Optional.ofNullable(credentialRepository.findByIdInAccount(id, user.getAccount()))
-                .orElseThrow(accessDenied(String.format("Access is denied: Credential not found by id: '%d'.", id)));
+    public void delete(Long id) {
+        Credential credential = Optional.ofNullable(
+                credentialRepository.findByIdAndOrganization(id, getDefaultOrg().getId(), accountPreferencesService.enabledPlatforms()))
+                .orElseThrow(accessDenied(String.format(ACCESS_DENIED_FORMAT_MESS_ID, id, getDefaultOrg().getName())));
         delete(credential);
     }
 
-    public void delete(String name, IdentityUser user) {
-        Credential credential = Optional.ofNullable(credentialRepository.findByNameInAccount(name, user.getAccount(), user.getUserId()))
-                .orElseThrow(accessDenied(String.format("Access is denied: Credential not found by name '%s'.", name)));
+    public void delete(String name) {
+        Credential credential = Optional.ofNullable(
+                credentialRepository.findActiveByNameAndOrgId(name, getDefaultOrg().getId(), accountPreferencesService.enabledPlatforms()))
+                .orElseThrow(accessDenied(String.format(ACCESS_DENIED_FORMAT_MESS_NAME, name, getDefaultOrg().getName())));
         delete(credential);
     }
 
-    public Credential update(Long id) {
-        return credentialAdapter.update(get(id));
+    @Override
+    public Credential delete(Credential credential) {
+        if (canDelete(credential)) {
```

Comment:
I suggest to use super.delete()

Reply:
the super.delete() do repository delete but we can't do that in the case of credentials since we archive them instead of delete